tests/unit/workflows/dotnet_clipackage/test_actions.py

```diff
@@ -42,6 +45,24 @@ def test_global_tool_update_failed(self):
         action = GlobalToolInstallAction(self.subprocess_dotnet)
         self.assertRaises(ActionFailedError, action.execute)
 
+    def test_global_tool_parallel(self):
+        executor = ThreadPoolExecutor()
+
+        actions = [
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+        ]
+
+        for action in actions:
+            executor.submit(action.execute)
+
+        executor.shutdown()
+
+        self.subprocess_dotnet.assert_has_calls(
+            [call.run(["tool", "install", "-g", "Amazon.Lambda.Tools", "--ignore-failed-sources"])]
+        )
+
```

Comment:
You can also maybe add a test where self.subprocess_dotnet raises a DotnetCLIExecutionError?

Reply:
This is already been covered by other tests above.

Reply:
Oops, missed those!